### PR TITLE
Extract a CheckboxInput component from onboarding flow

### PR DESCRIPTION
**Pull request by selfcontained**
## Description

Extracts the checkbox input used in the onboarding flow to a `CheckboxInput` component and also creates a `CheckboxInputField` that wraps a list of checkboxes in the same styling as other input type components do. This should allow us to use it as a field w/ a list of checkboxes, like the onboarding flow, or as a standalone checkbox like we do in other places.

image

```tsx
<CheckboxInputField label="My Label">
  <CheckboxInput value="1" label="First Option" checked={true} onChange={handler} />
  <CheckboxInput value="2" label="Second Option" checked={true} onChange={handler} />
  <CheckboxInput value="3" label="Third Option" checked={true} onChange={handler} />
</CheckboxInputField>
```

Other places in our UI use a `<CheckBox/>` component that I've referred to for some of the styles here, but plan on replacing those with this `<CheckboxInput/>` component for consistency. There's a few things to figure out w/ some of our font sizing, weights, and colors for form-related components, so I'll leave that as a separate task.

## Related Issue(s)
 
Relates to #16511

## How to test

* Run through the onboarding flow, and step 3 will have the checkboxes that use this component.
* https://bmh-checkb5001557072.preview.gitpod-dev.com/workspaces?onboarding=force

## Release Notes

```release-note
NONE
```

## Build Options:

- [ ] /werft with-github-actions
      Experimental feature to run the build with GitHub Actions (and not in Werft).
- [ ] leeway-no-cache
      leeway-target=components:all
- [ ] /werft no-test
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer Options</summary>

- [ ] with-ee-license
- [ ] with-slow-database
- [ ] with-dedicated-emulation
- [ ] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

**Comment by roboquat** [comment]
@selfcontained: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: components/dashboard/src/components/forms/InputField.tsx [hunk 1]
```diff
@@ -21,7 +22,7 @@ export const InputField: FunctionComponent<Props> = memo(({ label, id, hint, err
             {label && (
                 <label
                     className={classNames(
-                        "text-sm font-semibold dark:text-gray-400",
+                        "text-md font-bold dark:text-gray-400",
```
Comment (selfcontained) [inline components/dashboard/src/components/forms/InputField.tsx:25]: Slightly increasing size of input labels, and making them bold.

**Review by easyCZ — APPROVED** [review]
LGTM

/hold in case you want feedback from design

**Pull request merged**